WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -1026,32 +1026,74 @@ private extension NotificationsViewController {
+                                       comment: "Displayed in the Notifications Tab as a message, when the Unread Filter shows no notifications"),
+            .comment: NSLocalizedString("Join a conversation: comment on posts from blogs you follow.",
+                                        comment: "Displayed in the Notifications Tab as a message, when the Comments Filter shows no notifications"),
+            .follow: NSLocalizedString("Get noticed: comment on posts you've read.",
+                                       comment: "Displayed in the Notifications Tab as a message, when the Follow Filter shows no notifications"),
+            .like: NSLocalizedString("Get noticed: comment on posts you've read.",
+                                     comment: "Displayed in the Notifications Tab as a message, when the Likes Filter shows no notifications")
         ]
 
         let filter = Filter(rawValue: filtersSegmentedControl.selectedSegmentIndex) ?? .none
         return messageMap[filter] ?? String()
     }
 
-    var noResultsMessageText: String? {
-        let jetpackMessage = NSLocalizedString("Jetpack supercharges your self-hosted WordPress site.", comment: "Notifications message displayed when a self-hosted user is not connected to Jetpack")
-        return shouldDisplayJetpackMessage ? jetpackMessage : nil
-    }
-
     var noResultsAccessoryView: UIView? {
         return shouldDisplayJetpackMessage ? UIImageView(image: UIImage(named: "icon-jetpack-gray")) : nil
     }
 
     var noResultsButtonText: String? {
-        return shouldDisplayJetpackMessage ? NSLocalizedString("Learn more", comment: "") : nil
+        guard shouldDisplayJetpackMessage == false else {
+            return NSLocalizedString("Learn more", comment: "")
+        }
+
+        let buttonTitleMap: [Filter: String] = [
+            .none: NSLocalizedString("View Reader",
+                                     comment: "Displayed in the Notifications Tab as a button title, when there are no notifications"),
+            .unread: NSLocalizedString("New Post",
+                                       comment: "Displayed in the Notifications Tab as a button title, when the Unread Filter shows no notifications"),
+            .comment: NSLocalizedString("View Reader",
+                                        comment: "Displayed in the Notifications Tab as a button title, when there are no notifications"),
+            .follow: NSLocalizedString("View Reader",
+                                       comment: "Displayed in the Notifications Tab as a button title, when there are no notifications"),
+            .like: NSLocalizedString("View Reader",
+                                     comment: "Displayed in the Notifications Tab as a button title, when there are no notifications")
+        ]
+
+        let filter = Filter(rawValue: filtersSegmentedControl.selectedSegmentIndex) ?? .none
+        return buttonTitleMap[filter] ?? String()
```

Comment:
Note: (by all means, **no need to change anything**!!) I meant to convert this `map` mechanism, to actually, a `Filter property`. Main reason is... suppose:

```
enum Filter {
   //..
   var title: String {
      switch self { 
         case .none: return ...
      }

   }
}
```

By doing something like this, (if) a new filter gets added, we get free compiler warnings for missing case(s). Whereas... this old `map` mechanism keeps on building, and we need to figure out on our own all of the spots that must be updated.

Props to @frosty for that tip, i've adopted that practise a couple months ago... but couldn't get to this refactor!